gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/TopicId.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub;
+
+import static com.google.cloud.pubsub.spi.v1.PublisherApi.formatTopicName;
+import static com.google.cloud.pubsub.spi.v1.PublisherApi.parseProjectFromTopicName;
+import static com.google.cloud.pubsub.spi.v1.PublisherApi.parseTopicFromTopicName;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Identity for a Google PubSub topic. A {@code TopicId} object can be used to create subscriptions
+ * for topics, possibly residing on a different project.
```

Comment:
I think "that" or ", which" sounds nicer than just ",". 
Also, I think "belong to different" sounds better and in either case I think "project" needs to be plural.